### ISSUE TITLE
Add ARIA roles and attributes to the alert component

### ISSUE THREAD

**Issue by Veyfeyken** [a-ui/core_branding_scss#153]
### Problem

[molecules/molecules.alert](https://a-ui.github.io/core_branding_scss/#molecules-alert)

The current markup doesn't have the required ARIA roles and attributes to be accessible with a screenreader.

```html
<div class="m-alert">
  <button class="a-button-transparent has-icon m-alert__close"><span class="fa fa-close" aria-label="Close"></span></button>
  <h4 class="h5 u-margin-bottom-xs">Alert</h4>
  <p class="u-margin-bottom">Some explanation.</p>
  <div class="m-alert__actions">
    <button class="a-button-outline a-button--small">Cancel</button>
    <button class="a-button a-button--small">Got it</button>
  </div>
</div>
```

![20201109-16-14-57](https://user-images.githubusercontent.com/870049/98559294-ba460a80-22a6-11eb-9209-d4a18a343b53.jpg)

### Solution

* Place the aria-label="close" on the button instead of the child span.
* Add role="alertdialog" on the parent div
* Add aria-modal="true/false" on the parent div (partial support, consider making the rest of the document [inert](https://github.com/WICG/inert) when an element has aria-modal=true for more robust support)
* Add an aria-labelledby attribute on the parent div that refers to the title of the dialog
* Focus should be send to the alert when it becomes active. After closing it, focus should return to the element that had the focus before. 

See the ARIA Alert and Message Dialog pattern for more information and an example.

Add a note to the documentation which explains the necessity of focus management and how to handle it?

### Links and articles
* [WAI-ARIA Design Patterns and Widgets: Alert and Message Dialogs](https://www.w3.org/TR/wai-aria-practices-1.2/#alertdialog)
* [Rob Dodson: Managing focus for accessibility](https://dev.to/robdodson/managing-focus-64l)

**Comment by TriangleJuice** [comment]
Hey @Veyfeyken this issue should be fixed in the next iteration of our core branding (`v4.3.0`)

**Comment by Veyfeyken** [comment]
Only missing aria-modal="true" on the parent div. Or was it left out intentionally for some reason?

**Comment by TriangleJuice** [comment]
Hey @Veyfeyken I will add an accessibility note for developers that they should add it to the parent <div> if they use the alert as a popup, since an alert can also be used inline.
